pkg/kube/config/config.go

```diff
@@ -0,0 +1,66 @@
+/*
+Copyright 2024 The Crossplane Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+	http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+// Package config contains API types used by Crossplane providers interacting
+// with Kubernetes APIs.
+// +kubebuilder:object:generate=true
+package config
+
+import xpv1 "github.com/crossplane/crossplane-runtime/apis/common/v1"
+
+// IdentityType used to authenticate to the Kubernetes API.
+// +kubebuilder:validation:Enum=GoogleApplicationCredentials;AzureServicePrincipalCredentials;AzureWorkloadIdentityCredentials;UpboundToken
```

Comment:
@turkenh think here is bug 
```suggestion
// +kubebuilder:validation:Enum=GoogleApplicationCredentials;AzureServicePrincipalCredentials;AzureWorkloadIdentityCredentials;UpboundTokens
```

Reply:
🤦 fixed!